### PR TITLE
chore(acir): public block id 

### PR DESCRIPTION
**Pull request by guipublic**
# Description

## Summary of changes

Make the id of BlockId public so we can create BlockId from a number.

# Checklist

- [ ] I have tested the changes locally.
- [X] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` with default settings.
- [ ] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this PR to the issue(s) that it resolves.
- [X] I have reviewed the changes on GitHub, line by line.
- [X] I have ensured all changes are covered in the description.

# Additional context

We need to create the blocks in noir from the arrays

### CODE REVIEW

**Review by jfecher — COMMENTED** [review]
What functionality is this needed for?

File: acir/src/circuit/opcodes.rs [hunk 1]
```diff
@@ -7,7 +7,7 @@ use crate::BlackBoxFunc;
 use serde::{Deserialize, Serialize};
 
 #[derive(Clone, PartialEq, Eq, Serialize, Deserialize, Hash, Copy, Default)]
-pub struct BlockId(u32);
+pub struct BlockId(pub u32);
```
Comment (phated) [inline acir/src/circuit/opcodes.rs:10]: Would this be implemented with a `from<u32>` so the way it is stored isn't leaked to consumers/creators?

**Comment by guipublic** [comment]
> What functionality is this needed for?

The opcode is created by noir, so it needs to set the id. This is currently impossible as the constructor is private.

File: acir/src/circuit/opcodes.rs (hunk 1 above)
Comment (guipublic) [reply acir/src/circuit/opcodes.rs:10]: I don't see the problem to leak this.

Comment (guipublic) [reply acir/src/circuit/opcodes.rs:10]: No it won't change, I would not put anything else inside an id.

**Review by jfecher — APPROVED** [review]
Looks fine to me. We should usually be moving away from exposing internals of IDs in favor of keeping them opaque but it is indeed sometimes necessary to expose them. It is not worth keeping open such a simple PR, lets merge it.

**Pull request merged**